common/src/main/java/com/microsoft/identity/common/internal/net/HttpRequest.java

```diff
@@ -164,6 +168,9 @@ public static HttpResponse sendGet(final URL requestUrl, final Map<String, Strin
                         .putRequestIdHeader(requestHeaders == null ? null : requestHeaders.get(CLIENT_REQUEST_ID))
         );
 
+        final Map<String, String> telemetryHeaders = ServerTelemetry.getTelemetryHeaders();
```

Comment:
Same here - this is a violation of the separation of concerns: the network layer should not know about the telemetry layer, instead these headers should be passed in by the caller above

Reply:
Addressed